### ISSUE TITLE
A unique field on a component of a single-type leads to a faulty 'uniqueness check'

### ISSUE THREAD

**Issue by maxbethke** [strapi/strapi#16918]
### Required System information

- Node.js version: v16.17.1
- NPM version: 9.1.2
- Strapi version:
```
"dependencies": {
    "@strapi/plugin-i18n": "4.10.7",
    "@strapi/plugin-users-permissions": "4.10.7",
    "@strapi/strapi": "4.10.7",
    "mysql": "2.18.1"
  },
```
- Database: MySQL Server 5.7.41-1.el7
- Operating system: MacOS 12.6
- Is your project Javascript or Typescript: TS

### Describe the bug

A unique field on a component of a single-type leads to a faulty 'uniqueness check'

### Steps to reproduce the behaviour

- Create a new single-type with
   - a repeatable component on it
   - some other field, e.g. short text
- Create a text field on the component (short text, **unique**)
- Add a component to the single-type
- Save the single-type
- Put a value in the unique field of the component
- Save the single-type
- Edit any other field (just so you can hit save again)
- Save the single type

At this point error gets displayed saying the value in the unique field of the component is not unique, even though it is.

### Expected behavior
After the steps above, the single-type should be able save, because the unique field's value is unique.

### Idea
I might be very well wrong, but to me it looks like on the update() of the single-type, the 'uniqueness check' for the unique field does not only look at array of components that would result from update, but at the old data merged with the new data. Thus resulting in the error, because the unique value now exists twice, once in the old record of the edited component and once in the new record.

**Comment by rbsam176** [comment]
If anyone has any workarounds (other than not making the field unique) while this is being fixed it would be great to hear them, it's affecting our app significantly.

**Comment by jhoward1994** [comment]
Hi, thank you for the bug report. We are aware of this issue with unique fields and are actively working on it.

We also plan to extend the capabilities of unique fields in future, please see https://github.com/strapi/strapi/issues/12541#issuecomment-1503127208

**Comment by Kartik4152** [comment]
Any updates on this?

**Comment by DenuxPlays** [comment]
Also happens on collection-type.
Please update title!

**Comment by jhoward1994** [comment]
This issue has been fixed in v5 from 5.0.0-beta.13 onwards via https://github.com/strapi/strapi/pull/20427
